plugins/modules/job_invocation.py

```diff
@@ -0,0 +1,222 @@
+          - Defaults to a template parameter or global setting
+        type: str
+  feature:
+    description:
+      - Remote execution feature label that should be triggered, job template assigned to this feature will be used
+    type: str
+  command:
+    description:
+      - Command to be executed on host. Required for command templates
+    type: str
+  inputs:
+    description:
+      - Inputs to use
+    type: dict
+  recurrence:
+    description:
+      - Schedule a recurring job
+    type: dict
+    suboptions:
+      cron_line:
+        description:
+          - How often the job should occur, in the cron format
+        type: str
+      max_iteration:
+        description:
+          - Repeat a maximum of N times
+        type: int
+      end_time:
+        description:
+          - Perform no more executions after this time
+        type: str
+  scheduling:
+    description:
+      - Schedule the job to start at a later time
+    type: dict
+    suboptions:
+      start_at:
+        description:
+          - Schedule the job for a future time
+        type: str
+      start_before:
+        description:
+          - Indicates that the action should be cancelled if it cannot be started before this time.
+        type: str
+  concurrency_control:
+    description:
+      - Control concurrency level and distribution over time
+    type: dict
+    suboptions:
+      time_span:
+        description:
+          - Distribute tasks over given number of seconds
+        type: int
+      concurrency_level:
+        description:
+          - Maximum jobs to be executed at once
+        type: int
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state
+  - theforeman.foreman.foreman.taxonomy
+'''
+
+EXAMPLES = '''
+
+- name: "Run remote command on a single host once"
+  job_invocation:
+    search_query: "name ^ (foreman.example.com)"
+    command: 'ls'
+    job_template: "Run Command - SSH Default"
+    ssh:
+      effective_user: "tester"
+
+- name: "Run ansible commad on active hosts once a day"
+  job_invocation:
+    bookmark: 'active'
+    command: 'pwd'
+    job_template: "Run Command - Ansible Default"
+    recurrence:
+      cron_line: "30 2 * * *"
+    concurrency_control:
+      concurrency_level: 2
+'''
+
+RETURN = ''' # '''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import (
+    ForemanTaxonomicEntityAnsibleModule,
+)
+
+ssh_foreman_spec = {
+    'effective_user': dict(),
+}
+
+recurrence_foreman_spec = {
+    'cron_line': dict(),
+    'max_iteration': dict(type='int'),
+    'end_time': dict(),
+}
+
+scheduling_foreman_spec = {
+    'start_at': dict(),
+    'start_before': dict(),
+}
+
+concurrency_control_foreman_spec = {
+    'time_span': dict(type='int'),
+    'concurrency_level': dict(type='int'),
+}
+
+
+class ForemanJobInvocationModule(ForemanTaxonomicEntityAnsibleModule):
```

Comment:
My comment in https://github.com/theforeman/foreman-ansible-modules/pull/864/files#r473665687 still stands. I think this module should not support `state=absent` or updating an existing invocation because the API does not support that.
At the same time having an *entity* module helps here, as we do *create* an entity when we create the invocation.
We should just not support deletion or updating. The following is probably a good start: use `ForemanTaxonomicAnsibleModule` here and rewrite the `module.run()` to smth like:

```python
entity = module.lookup_entity('entity')
if entity is None:
    module.ensure_entity('job_invocations', module.foreman_params, entity)
else:
    module.fail_json(msg="You cannot update an existing job_invocation")
```

That still requires `lookup_entity` to actually work, which it currently does not. See my comment about `search_by`

Reply:
We have `ForemanStatelessEntityAnsibleModule` for those cases.

Reply:
@evgeni, @mdellweg  uh oh, went with Evgeni's suggestion already, should I rewrite with ForemanStatelessEntityAnsibleModule ?

Reply:
StatelessEntity can be updated, which a job invocation can not, so I'd think the current way is more correct.

But I also forgot about StatelessEntity when originally writing the above comment. :woman_shrugging: